src/index/DeltaTriples.cpp

```diff
@@ -31,37 +33,54 @@ LocatedTriples::iterator& DeltaTriples::LocatedTripleHandles::forPermutation(
 void DeltaTriples::clear() {
   triplesInserted_.clear();
   triplesDeleted_.clear();
+  internalTriplesInserted_.clear();
+  internalTriplesDeleted_.clear();
   ql::ranges::for_each(locatedTriples(), &LocatedTriplesPerBlock::clear);
+  ql::ranges::for_each(internalLocatedTriples_, &LocatedTriplesPerBlock::clear);
```

Comment:
if the proposed `State` also has a `clear  function, then this boils down to two calls.